From 11a0d1547e3fff4bcbd8b0ecc1bbfc4300c091ee Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20Br=C3=BCns?= <stefan.bruens@rwth-aachen.de>
Date: Tue, 20 Sep 2022 01:23:25 +0200
Subject: [PATCH] Use ILP32 model on 32 bit ARM

GCC on ARM does not define __ILP32__ or similar, so detect it explicitly.

Fixes #70.
---
 api/vpl/mfxdefs.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/api/vpl/mfxdefs.h b/api/vpl/mfxdefs.h
index 35bfb0fc..f9b468ea 100644
--- a/api/vpl/mfxdefs.h
+++ b/api/vpl/mfxdefs.h
@@ -68,7 +68,7 @@ extern "C"
     #define MFX_PACK_BEGIN_STRUCT_W_PTR()    MFX_PACK_BEGIN_X(4)
     #define MFX_PACK_BEGIN_STRUCT_W_L_TYPE() MFX_PACK_BEGIN_X(8)
 /* 32-bit ILP32 data model Linux* */
-#elif defined(__ILP32__)
+#elif defined(__ILP32__) || defined(__arm__)
     #define MFX_PACK_BEGIN_STRUCT_W_PTR()    MFX_PACK_BEGIN_X(4)
     #define MFX_PACK_BEGIN_STRUCT_W_L_TYPE() MFX_PACK_BEGIN_X(4)
 #else
